internal/node/node_repositories.bzl

```diff
@@ -442,45 +402,33 @@ def node_repositories(package_json, node_version=DEFAULT_NODE_VERSION, yarn_vers
   # Windows users need sh_binary wrapped as an .exe
   check_bazel_version("0.5.4")
 
-  # If the caller already executed node_download_runtime or node_local_runtime, and correctly used the "nodejs"
-  # repo name, then we can skip this step.
-  if "nodejs" not in native.existing_rules():
-    if node_version in NODE_REPOSITORIES:
-      node_download_runtime(
-          name = "nodejs",
-          version = node_version,
-          packages = NODE_REPOSITORIES[node_version],
-          package_json = package_json,
-          preserve_symlinks = preserve_symlinks,
-      )
-    else:
-      fail("Unknown NodeJS version {}".format(node_version))
-
-  # If the caller already executed yarn_download or yarn_local, and correctly used the "yarn" repo name, then we can
-  # skip this step.
-  if "yarn" not in native.existing_rules():
-    if yarn_version in YARN_REPOSITORIES:
-      filename, strip_prefix, sha256 = YARN_REPOSITORIES[yarn_version]
-      yarn_download(
-        name = "yarn",
-        version = yarn_version,
-        filename = filename,
-        strip_prefix = strip_prefix,
-        sha256 = sha256,
-        package_json = package_json,
-        preserve_symlinks = preserve_symlinks,
-      )
-    else:
-      fail("Unknown Yarn version {}".format(yarn_version))
+  _nodejs_repo(
+    name = "nodejs",
+    package_json = package_json,
+    node_version = node_version,
+    yarn_version = yarn_version,
+    node_path = node_path,
+    yarn_path = yarn_path,
+    node_repositories = node_repositories,
+    yarn_repositories = yarn_repositories,
+    node_urls = node_urls,
+    yarn_urls = yarn_urls,
+    preserve_symlinks = preserve_symlinks,
+  )
+
+  _yarn_repo(
+    name = "yarn",
+    package_json = package_json,
+  )
 
   yarn_install(
-      name = "build_bazel_rules_nodejs_npm_install_deps",
-      package_json = "@build_bazel_rules_nodejs//internal/npm_install:package.json",
-      yarn_lock = "@build_bazel_rules_nodejs//internal/npm_install:yarn.lock",
+    name = "build_bazel_rules_nodejs_npm_install_deps",
+    package_json = "@build_bazel_rules_nodejs//internal/npm_install:package.json",
+    yarn_lock = "@build_bazel_rules_nodejs//internal/npm_install:yarn.lock",
   )
 
   yarn_install(
-      name = "build_bazel_rules_nodejs_rollup_deps",
-      package_json = "@build_bazel_rules_nodejs//internal/rollup:package.json",
-      yarn_lock = "@build_bazel_rules_nodejs//internal/rollup:yarn.lock",
+    name = "build_bazel_rules_nodejs_rollup_deps",
```

Comment:
nit: 4 space indent is correct for continuation

Reply:
👍